clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/AbstractStackUpdateOpenstackAtomicOperation.groovy

```diff
@@ -67,20 +67,32 @@ abstract class AbstractStackUpdateOpenstackAtomicOperation implements AtomicOper
       //we need to store subtemplate in asg output from create, as it is required to do an update and there is no native way of
       //obtaining it from a stack
-      List<Map<String, Object>> outputs = stack.outputs
-      String subtemplate = outputs.find { m -> m.get("output_key") == ServerGroupConstants.SUBTEMPLATE_OUTPUT }.get("output_value")
-      String memberTemplate = outputs.find { m -> m.get("output_key") == ServerGroupConstants.MEMBERTEMPLATE_OUTPUT }.get("output_value")
-      task.updateStatus phaseName, "Successfully fetched server group $description.serverGroupName"
-
-      //get the current template from the stack
-      task.updateStatus phaseName, "Fetching current template for server group $description.serverGroupName"
-      String template = provider.getHeatTemplate(description.region, stack.name, stack.id)
-      task.updateStatus phaseName, "Successfully fetched current template for server group $description.serverGroupName"
-
-      //update stack
-      task.updateStatus phaseName, "Updating server group $stack.name"
-      provider.updateStack(description.region, stack.name, stack.id, template, [(ServerGroupConstants.SUBTEMPLATE_FILE): subtemplate, (ServerGroupConstants.MEMBERTEMPLATE_FILE): memberTemplate], buildServerGroupParameters(stack), stack.tags)
-      task.updateStatus phaseName, "Successfully updated server group $stack.name"
+      String resourceFileName = stack.parameters?.get(ServerGroupConstants.SUBTEMPLATE_FILENAME)
+
+      if (resourceFileName) {
+        List<Map<String, Object>> outputs = stack.outputs
+        String resourceSubtemplate = outputs.find { m -> m.get("output_key") == ServerGroupConstants.SUBTEMPLATE_OUTPUT }.get("output_value")
+        String memberTemplate = outputs.find { m -> m.get("output_key") == ServerGroupConstants.MEMBERTEMPLATE_OUTPUT }.get("output_value")
+        task.updateStatus phaseName, "Successfully fetched server group $description.serverGroupName"
+
+        //get the current template from the stack
+        task.updateStatus phaseName, "Fetching current template for server group $description.serverGroupName"
+        String template = provider.getHeatTemplate(description.region, stack.name, stack.id)
+        task.updateStatus phaseName, "Successfully fetched current template for server group $description.serverGroupName"
+
+        Map<String, String> templateMap = [(resourceFileName): resourceSubtemplate]
+        if (memberTemplate) {
+          templateMap <<  [(ServerGroupConstants.MEMBERTEMPLATE_FILE): memberTemplate]
+        }
+
+        //update stack
+        task.updateStatus phaseName, "Updating server group $stack.name"
+        provider.updateStack(description.region, stack.name, stack.id, template, templateMap, buildServerGroupParameters(stack), stack.tags)
+        task.updateStatus phaseName, "Successfully updated server group $stack.name"
+      } else {
+        task.updateStatus phaseName, "Missing resource filename in parameters list"
```

Comment:
If there are many parallel operations being processed, it may be difficult to troubleshoot without these "Missing..." error messages having more detailed information.